code/model/StateMachineFactory.php

```diff
@@ -13,20 +13,41 @@ public function forDNDeployment(DNDeployment $obj) {
 				DNDeployment::STATE_NEW => ['type' => StateInterface::TYPE_INITIAL],
 				DNDeployment::STATE_SUBMITTED => ['type' => StateInterface::TYPE_NORMAL],
 				DNDeployment::STATE_INVALID => ['type' => StateInterface::TYPE_NORMAL],
+				DNDeployment::STATE_APPROVED => ['type' => StateInterface::TYPE_NORMAL],
+				DNDeployment::STATE_REJECTED => ['type' => StateInterface::TYPE_NORMAL],
 				DNDeployment::STATE_QUEUED => ['type' => StateInterface::TYPE_NORMAL],
 				DNDeployment::STATE_DEPLOYING => ['type' => StateInterface::TYPE_NORMAL],
 				DNDeployment::STATE_ABORTING => ['type' => StateInterface::TYPE_NORMAL],
 				DNDeployment::STATE_COMPLETED => ['type' => StateInterface::TYPE_FINAL],
 				DNDeployment::STATE_FAILED => ['type' => StateInterface::TYPE_FINAL],
 			],
 			'transitions' => [
+				DNDeployment::TR_NEW => ['from' => [DNDeployment::STATE_SUBMITTED], 'to' => DNDeployment::STATE_NEW],
 				DNDeployment::TR_SUBMIT => ['from' => [DNDeployment::STATE_NEW], 'to' => DNDeployment::STATE_SUBMITTED],
-				DNDeployment::TR_QUEUE => ['from' => [DNDeployment::STATE_SUBMITTED], 'to' => DNDeployment::STATE_QUEUED],
+				DNDeployment::TR_QUEUE => [
+					'from' => [
+						DNDeployment::STATE_SUBMITTED,
```

Comment:
what does submitted mean now? Does this mean you can deploy while waiting for approval?